### ISSUE TITLE
Table schema design issue with comments

### ISSUE THREAD

**Issue by serkanatpaystack** [Arinzeokeke/star-wars-api#3]
I see that you've included both movie id and movie url in the comments table.

Does this not both duplicate information, causing a confusion around the single source of truth and potentially a future discrepancy if movie url structure changes?

Furthermore, why do we need the movie url in this, as relevant to the comments?

**Comment by Arinzeokeke** [comment]
> I see that you've included both movie id and movie url in the comments table.
> 
> Does this not both duplicate information, causing a confusion around the single source of truth and potentially a future discrepancy if movie url structure changes?
>
> Furthermore, why do we need the movie url in this, as relevant to the comments?

You're right,  the `movie_id` really should be the only unique identifier and having the `movie_url` stored too could lead to confusion.
But there's a reason I did that. While I was going through the docs and calling their API, I found that swapi API responses don't include the `id` primary key.

Normally, this information can be gotten from the `movie_id` path parameter in our own request URL.
However, specifically in the case of `GET /movies` (fetching all the movies along with its comment count), we do not have access to the `movie_id` from the URL. 

In this case, for each movie in the list, we need a unique identifier to determine their comments and we don't have access to the movie ID from the movie object.

So, I chose to also store the `movie_url` as its another identifier that could be used to fetch specific comments for a movie using the `movie_url` as the identifier as, in this scenario,  we don't have access to the movie's `id` unique identifier. 

I tried to find another way around that, including parsing the `url` field and getting the `movie_id` substring from it. But I settled on also keeping track of the URL.

**Comment by serkanatpaystack** [comment]
The remote `/films` endpoint returns something labeled `episode_id` already, which you could have cached and used as your unique identifier as opposed to the url. This way, there's at least much less chance of this changing. The url can change if the base url changes for instance.

Nonetheless, going the extra mile to get the movie id out from the url would have been a preferable choice rather than leaving the app susceptible to data integrity issues.
